diff --git a/tensorflow/lite/BUILD b/tensorflow/lite/BUILD
index f43b8fd4c1..811f3aef05 100644
--- a/tensorflow/lite/BUILD
+++ b/tensorflow/lite/BUILD
@@ -165,7 +165,7 @@ cc_library(
             "mmap_allocation_disabled.cc",
         ],
         "//conditions:default": [
-            "mmap_allocation.cc",
+            "mmap_allocation_disabled.cc",
         ],
     }),
     hdrs = [
diff --git a/tensorflow/lite/kernels/BUILD b/tensorflow/lite/kernels/BUILD
index 95f554adcf..686ac97a53 100644
--- a/tensorflow/lite/kernels/BUILD
+++ b/tensorflow/lite/kernels/BUILD
@@ -25,7 +25,7 @@ EXTRA_EIGEN_COPTS = select({
         "/DEIGEN_HAS_C99_MATH",
         "/DEIGEN_AVOID_STL_ARRAY",
     ],
-    "//conditions:default": ["-Wno-error=reorder"],
+    "//conditions:default": ["-Wno-error=reorder -DEIGEN_NO_CPUID"],
 })
 
 cc_test(
diff --git a/tensorflow/lite/kernels/internal/round.h b/tensorflow/lite/kernels/internal/round.h
index d102d37933..dc5679db22 100644
--- a/tensorflow/lite/kernels/internal/round.h
+++ b/tensorflow/lite/kernels/internal/round.h
@@ -31,7 +31,7 @@ inline double TfLiteRound(const double x) { return ::round(x); }
 #else
 template <class T>
 inline T TfLiteRound(const T x) {
-  return std::round(x);
+  return ::round(x);
 }
 #endif
 
diff --git a/tensorflow/lite/nnapi/BUILD b/tensorflow/lite/nnapi/BUILD
index 8b85a5f118..0ad524c3e5 100644
--- a/tensorflow/lite/nnapi/BUILD
+++ b/tensorflow/lite/nnapi/BUILD
@@ -12,7 +12,7 @@ cc_library(
     ],
     linkopts = select({
         "//tensorflow:windows": [],
-        "//conditions:default": ["-ldl"],
+        "//conditions:default": [],
     }),
 )
 
@@ -26,7 +26,7 @@ cc_library(
             "nnapi_implementation_disabled.cc",
         ],
         "//conditions:default": [
-            "nnapi_implementation.cc",
+            "nnapi_implementation_disabled.cc",
         ],
     }),
     hdrs = [
@@ -34,13 +34,13 @@ cc_library(
     ],
     linkopts = select({
         "//tensorflow:windows": [],
-        "//conditions:default": ["-ldl"],
+        "//conditions:default": [],
     }) + select({
         "//tensorflow:android": [],
         "//tensorflow:macos": [],
         "//tensorflow:ios": [],
         "//tensorflow:windows": [],
-        "//conditions:default": ["-lrt"],
+        "//conditions:default": [],
     }),
     deps = [
         "//tensorflow/lite/nnapi:nnapi_lib",
